tests/test_agent_discovery.py

```diff
@@ -0,0 +1,172 @@
+"""Tests for agent discovery functionality."""
+
```

Comment:
not sure if we need these but I'm anticipating coverage will require them from me so I'll leave it here for now